lib/active_merchant/billing/gateways/trans_first_transaction_express.rb

```diff
@@ -537,14 +537,16 @@ def add_contact(doc, fullname, options)
           doc["v1"].title options[:title] if options[:title]
 
           if (billing_address = options[:billing_address])
-            doc["v1"].phone do
-              doc["v1"].type (options[:phone_number_type] || "4")
-              doc["v1"].nr billing_address[:phone].gsub(/\D/, '') if billing_address[:phone]
+            if billing_address[:phone]
+              doc["v1"].phone do
+                doc["v1"].type (options[:phone_number_type] || "4")
+                doc["v1"].nr billing_address[:phone].gsub(/\D/, '')
+              end
             end
             doc["v1"].addrLn1 billing_address[:address1]
             doc["v1"].addrLn2 billing_address[:address2] if billing_address[:address2]
-            doc["v1"].city billing_address[:city]
-            doc["v1"].state billing_address[:state]
+            doc["v1"].city billing_address[:city] if billing_address[:city]
+            doc["v1"].state billing_address[:state] if billing_address[:state]
             doc["v1"].zipCode billing_address[:zip]
```

Comment:
Is zip expected to always be sent? Wondering if we should follow this same pattern?